scripts/utils/generate_ts.sh

```diff
@@ -113,6 +128,20 @@ for branch in $(git branch -r | grep origin/releases); do
   done
 done
 
+# When creating translation files for l10n repo, remove any addon-specific files that use
+# shared strings, as they are translated via `strings.yaml`, which is set up for translation in
+# `build.py` (which like the addon ts files, is created above when addon's cmake calls `build.py`)"
+if [ $KEEP_ALL_TS_FILES == 0 ]; then
+  print Y "Checking for .ts files using shared strings"
+  for ts_file in ./addon_ts/*
+  do
+    if grep vpn.commonString $ts_file > /dev/null; then
```

Comment:
This is not reliable. The FIrefox extension addon uses shared strings, but no strings with `vpn.commonString` IDs.
https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/addons/message_try_firefox_extension/manifest.json

Can we use the ts_file to go back to the addon manifest, and check for `"usesSharedStrings": true`?

Reply:
Worth noting that I don't see the code going through `message_try_firefox_extension` for some reason.

Reply:
> Worth noting that I don't see the code going through `message_try_firefox_extension` for some reason (not sure if it's because my system is only partially set up).

I think this was indeed due to my system.

Something like this is clunky but could work?

```bash
if [ $KEEP_ALL_TS_FILES == 0 ]; then
  print Y "Checking for addons using shared strings"
  for ts_file in ./addon_ts/*
  do
    # Use the addon name from the .ts file to build a path to the manifest
    manifest_file="$(pwd)/addons/$(basename $ts_file .ts)/manifest.json"
    if [[ -f $manifest_file ]]; then
      uses_shared_strings=$(jq '.message.usesSharedStrings // false' < $manifest_file)
      if [[ "$uses_shared_strings" == "true" ]]; then
        print G "Deleting file because found shared strings: $ts_file"
        rm $ts_file
      fi
    else
      print R "Manifest file not found: $manifest_file"
    fi
  done
fi
```

Not sure why I'm getting an error on `Manifest file not found: mozilla-vpn-client/addons/message_update_v2.24/manifest.json`. I can see the .ts file being generated, but that addon doesn't exist in `/addons` 🤔

Reply:
I want to `die` if the manifest file isn't found, because it should always be there. 

To get this working, I had to have similar logic in two spots - doing it once at the start (on the branch the user was on when calling the script), then once again each time we loop through the release branches. I really do not like this pattern, but it's in line with how this file currently handles things - it does a bunch of work on the first branch, then essentially duplicates most of those calls (and adds in calls that combine the various files) for each release branch in the for loop.

Reply:
Good catch, and I'm sorry - my fix neglects a situation where none of the shared strings are a common string.

Agreed testing this is somewhat awful, thanks for all your support with it.

To answer another question: Shared strings for messages are indeed all or nothing. Either all strings come from that file, or we use the old fashioned way. (My intention would be to use only the new shared file moving forward.)